Source/Core/DolphinQt2/GameList/GameListModel.cpp

```diff
@@ -138,6 +138,63 @@ int GameListModel::columnCount(const QModelIndex& parent) const
   return NUM_COLS;
 }
 
+bool GameListModel::ShouldDisplayGameListItem(const QModelIndex& index) {
+  QSharedPointer<GameFile> game = m_games[index.row()];
+
+
+  const bool show_platform = [&game] {
+    switch (game->GetPlatformID())
+    {
+    case DiscIO::Platform::GAMECUBE_DISC:
+      return SConfig::GetInstance().m_ListGC;
+    case DiscIO::Platform::WII_DISC:
+      return SConfig::GetInstance().m_ListWii;
+    case DiscIO::Platform::WII_WAD:
+      return SConfig::GetInstance().m_ListWad;
+    case DiscIO::Platform::ELF_DOL:
+      return SConfig::GetInstance().m_ListElfDol;
+    default:
+      return false;
+    }
+  }();
```

Comment:
Rather than using a lambda for control-flow, it'd be more desirable to write it out, like so:

```c++
bool show_platform;
switch (game->GetPlatformID())
{
case DiscIO::Platform::GAMECUBE_DISC:
  show_platform = SConfig::GetInstance().m_ListGC;
case DiscIO::Platform::WII_DISC:
  show_platform = SConfig::GetInstance().m_ListWii;
case DiscIO::Platform::WII_WAD:
  show_platform = SConfig::GetInstance().m_ListWad;
case DiscIO::Platform::ELF_DOL:
  show_platform = SConfig::GetInstance().m_ListElfDol;
default:
  show_platform = false;
}